extractor/src/main/java/org/schabi/newpipe/extractor/utils/DonationLinkHelper.java

```diff
@@ -31,10 +31,10 @@ public static DonationService getDonationServiceByLink(String link) throws Malfo
 
     public static AffiliateService getAffiliateServiceByLink(String link) throws MalformedURLException {
         URL url = new URL(fixLink(link));
-        switch (url.getHost()) {
-            case "amzn.to": return AffiliateService.AMAZON;
-            default: return AffiliateService.NO_AFFILIATE;
+        if ("amzn.to".equals(url.getHost())) {
+            return AffiliateService.AMAZON;
         }
+        return AffiliateService.NO_AFFILIATE;
```

Comment:
Is this really better? I like the switch version and if more cases are added that one is cleaner

Reply:
changed